### PR TITLE
Add documentation for reviewers and a checklist for PRs

### PR DESCRIPTION
**Pull request by smoia**
Closes none, depends on #313.

## Proposed Changes

  - Update the PR template to have a checklist of steps that should be completed before promoting a draft PR to "Open PR" (Or before creating one) and more comments.
  - Add a documentation section on reviewing
  - Add a documentation section on Main Reviewers

### CODE REVIEW

**Comment by smoia** [comment]
I know we didn't solve the issues with `auto` yet, but I'm going to unpause this already.
@physiopy/phys2bids anyone wants to have a look?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (eurunuela) [inline .github/PULL_REQUEST_TEMPLATE.md:3]:
```suggestion
<!-- If you just want to reference them without closing them, you can add something like "References #112" -->
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (eurunuela) [inline .github/PULL_REQUEST_TEMPLATE.md:6]:
```suggestion
<!-- Add a short description of the PR content here-->
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (eurunuela) [inline .github/PULL_REQUEST_TEMPLATE.md:24]:
```suggestion
<!-- If this section is not clear, please read this part of the docs: https://phys2bids.readthedocs.io/en/latest/contributorfile.html#pr -->
```

File: docs/contributing.rst
Comment (eurunuela) [inline docs/contributing.rst:16]:
```suggestion
3. If you intend to contribute code and/or use the ``physiopy`` packages in any way, check that you have ``git`` and ``pip`` installed on your system. Then install ``phys2bids`` as a developer. This will let you run the program with the latest modifications, without requiring to re-install it every time.
```

**Review by eurunuela — CHANGES_REQUESTED** [review]
I've some tiny suggestions.

**Comment by RayStick** [comment]
Good initiative - this will be helpful. 

Do you think including something about the timeline of a review would be needed? (Sorry if I've missed this somewhere)

I.e. "I know this PR needs to be merged before we can work on X", or "it would be great to get this done by X day". Then a reviewer that is assigned to a few PRs might know how best to spend their time, or they say they don't think they can get it done in that time frame (yes, I might be thinking personally here!). I guess that's kind of what the milestones are for, right?

**Comment by RayStick** [comment]
I suggested some changes to your branch @smoia. Other than that, LGTM

**Review by RayStick — CHANGES_REQUESTED** [review]
I suggested some changes to your branch @smoia. Other than that, LGTM

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tsalo) [inline .github/PULL_REQUEST_TEMPLATE.md:31]: You might need to change that based on #324 😆

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tsalo) [inline .github/PULL_REQUEST_TEMPLATE.md:32]:
```suggestion
- [ ] My code respects the adopted style, especially linting conventions.
```
(or `linter` I suppose, if the point is about the settings for the linter)

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tsalo) [inline .github/PULL_REQUEST_TEMPLATE.md:36]:
```suggestion
- [ ] Please, comment on my PR while it's a draft and give me feedback on the development!
```

File: docs/contributorfile.rst
Comment (tsalo) [inline docs/contributorfile.rst:209]:
```suggestion
- If your contribution contains code or tests, you ran and passed all of the tests locally with `pytest`.
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (smoia) [inline .github/PULL_REQUEST_TEMPLATE.md:30]:
```suggestion
- [ ] My contribution is harmonious with the rest of the code: I'm not introducing repetitions.
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (smoia) [inline .github/PULL_REQUEST_TEMPLATE.md:31]:
```suggestion

```

**Comment by tsalo** [comment]
One thing I didn't see (but might have missed) was something about scope. I personally have a habit of letting my PRs sprawl. Stuff like fixing typos outside of the code being targeted by the PR and such should probably be explicitly warned against. Apologies if this is in there, but I missed it.

**Comment by smoia** [comment]
I went through the changes, accepted all of them (apologies @RayStick , I missed the PR to my branch!) and added a sentence to invite contributors to narrow the scope of PRs considering @tsalo's point above.
Let me know if there is anything more to do!

**Review by RayStick — APPROVED** [review]
Great. When others review, I will do all the final checks as main reviewer, and merge if it all looks good.

File: docs/contributorfile.rst
Comment (tsalo) [inline docs/contributorfile.rst:265]:
```suggestion
    - Checks that all elements of this document were respected, especially the part about `Pull Requests <#pr>`_.
```

File: docs/contributorfile.rst
Comment (tsalo) [inline docs/contributorfile.rst:244]: To clarify, does this mean that reviewers shouldn't use the code suggestion tool (like you probably see throughout my review) or that they shouldn't directly commit to the contributor's branch? I personally like the code suggestion tool (obviously), so I'd like to allow reviewers to use it as needed.

File: docs/contributorfile.rst
Comment (tsalo) [inline docs/contributorfile.rst:272]:
```suggestion
- Double-checks that the title is clear and the labels are correct to trigger an appropriate ``auto`` release - feel free to change them.
```

File: docs/contributorfile.rst
Comment (tsalo) [inline docs/contributorfile.rst:271]:
```suggestion
    - Checks that the appropriate issues have been opened to address the lack of documentation or tests (1 issue per item).
```

File: docs/contributorfile.rst
Comment (tsalo) [inline docs/contributorfile.rst:274]:
```suggestion
- After the PR has been merged and a new release has been triggered, checks that:
```

File: docs/contributorfile.rst
Comment (tsalo) [inline docs/contributorfile.rst:199]: This confuses me a little. It implies to me that one set of changes (e.g., adding a new metric to `phys2denoise`), would involve three PRs: one to add the metric, one to add documentation about the metric, and one to add tests for the metric. The last one clearly falls under the first PR based on the rest of this document, but I don't think it's clear from context if contributors should separate the PR with the new metric+testing from another PR with documentation about the metric. Personally, I think all of it should be done in one PR in that situation.

**Review by tsalo — CHANGES_REQUESTED** [review]
I mostly have nitpicky wording requests that you can take or leave, but there are a couple of questions in there.

File: docs/contributorfile.rst
Comment (eurunuela) [inline docs/contributorfile.rst:285]: Should we wait for the discussion going on at #324 before merging?

**Review by eurunuela — COMMENTED** [review]
I'm happy with the PR but think @tsalo 's suggestions are good and should be addressed first.

I also have a question regarding #324

File: docs/contributorfile.rst
Comment (tsalo) [inline docs/contributorfile.rst:135]: This is a minor thing, but it would be awesome if the labels themselves could be replaced with badges.

Comment (tsalo) [reply docs/contributorfile.rst:135]: Here's an example (built [here](https://shields.io/category/issue-tracking)): https://img.shields.io/github/labels/physiopy/phys2bids/Documentation?style=plastic

Comment (eurunuela) [reply docs/contributorfile.rst:135]: > This is a minor thing, but it would be awesome if the labels themselves could be replaced with badges.

You mean in the bullet list?

Comment (tsalo) [reply docs/contributorfile.rst:135]: Yeah, just in the documentation.

Comment (tsalo) [reply docs/contributorfile.rst:135]: And the badges could probably link to the list of issues with each label, if we want.

Comment (eurunuela) [reply docs/contributorfile.rst:135]: That would be very convenient actually! I like that.

File: docs/contributorfile.rst
Comment (smoia) [reply docs/contributorfile.rst:244]: That's a practice I liked a lot and that I tried to pick up from @rmarkello: he never used the suggestion tools, opting for explaining the changes to the code he meant to apply and waiting for the PR author to do them instead - with the exception of rewording or typos or minimal things.
If you don't agree with this practice, we can reword it so that it's only against direct commits. I know that the suggestion tool make things waaaaaaaaaay faster, but for learning purposes is not the best.

File: docs/contributorfile.rst
Comment (smoia) [reply docs/contributorfile.rst:199]: The reasoning is two(three?)fold.
First, if we want to use the automatic `all-contributors` plugin for `auto`, if reviewers use the "suggestion" tool they are interpreted as co-authors of the PR. For instance, this PR is labelled "internal" and "documentation", so all reviewers I accepted suggestion from will get the contribution badge for `documentation` and `infrastructure`. Normally it doesn't matter so much because we tend to review the types of PR that match our previous contribution, but I already have to clean up a bit the contribution table.
Second, since the main focus of a `code` PR is code changes (and not tests or documentations, that have to be done but are _secondary_ in that context), even if the PR contains changes to the tests and documentation I would only label it with one of the two code-related label (majormod and minormod), since the code change is the real scope of the PR. This is also related to the fact that all-contributors has a binary nature (either you contributed in a field or you didn't). I find slightly unfair that two contributors have the same badge (e.g. `test`), but one made almost all the tests and the other two small changes. This is partially solved by the number and size of the PRs labelled in a certain way, so having clear, distinct PRs helps adding a metric of contribution.
Bonus reason: we could _not_ use the all-contributors plugin for `auto`, but most of the time it helps adding contributions impartially and it really helps keeping track of most of the contributions.

File: docs/contributorfile.rst
Comment (smoia) [reply docs/contributorfile.rst:285]: Even if we adopt an opinionated code formatter, I wouldn't force the contributors to adopt it during their coding or documentation writing. In fact, one of the reasons to adopt an opinionated code formatter is to _not have to respect any opinionated code format during coding_. Most of the time it's sufficient that the code respects *flake8* conventions, so that the changes introduced by opinionated code formatters do not introduce an incredible amount of differences compared to the submitted code - although it's true that it becomes less essential. Since we're going to have `black` run during PR merge (I guess?), I would keep this section as it is.

File: docs/contributorfile.rst
Comment (smoia) [reply docs/contributorfile.rst:135]: Yeah I wondered how did you do it in `tedana`, cause I definitely wanted to do it here!
I wouldn't stall this PR for this reason, but we can definitely open an issue about it!

Comment (smoia) [reply docs/contributorfile.rst:135]: (Also I would use the _flat_ style rather than the plastic one)

**Comment by smoia** [comment]
Ok, addressed most of the suggestions and questions!
However it might be worth addressing those questions during an upcoming meeting. What do you think?

File: docs/contributorfile.rst
Comment (eurunuela) [reply docs/contributorfile.rst:244]: Maybe we could make an exception with newbies and make it educational for them. Or if one of us would like an extended description of the suggestion, we can just ask for it!

File: docs/contributorfile.rst
Comment (eurunuela) [reply docs/contributorfile.rst:199]: Gotcha! Still, I think tests could go with code changes into the same PR. If we make two different PRs we will increase our version with each of them, which does not make sense to me if the PRs are 100% related like code + tests.

File: docs/contributorfile.rst
Comment (eurunuela) [reply docs/contributorfile.rst:244]: To me this PR is ready to be merged once we figure this part out.

Comment (RayStick) [reply docs/contributorfile.rst:244]: I don't _think_ I have used the code suggestion tool before. I normally just make comments, or put in a PR to the author's branch, with my suggested changes. I guess it would be good to be clear about what we prefer, if anything.

File: docs/contributorfile.rst
Comment (RayStick) [reply docs/contributorfile.rst:199]: I think it might make sense to include tests alongside the code, in the same PR, as well. 

And if a contributor writes the code, but is not sure about how to write the tests, that is when they can make a draft PR and ask for help from other people to include tests in their PR?

**Comment by RayStick** [comment]
It seems we are waiting to merge this, after discussion in the next meeting on some of the topics raised, is that correct?

File: docs/contributorfile.rst
Comment (smoia) [reply docs/contributorfile.rst:244]: I'm going to rewrite this point in the next commit!

File: docs/contributorfile.rst
Comment (smoia) [reply docs/contributorfile.rst:199]: I'm going to update after the discussion in the call!

**Comment by smoia** [comment]
@tsalo @RayStick @eurunuela I updated a couple of sentences to reflect the discussion we had here and during the last meeting.

Please have a look!

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (eurunuela) [inline .github/PULL_REQUEST_TEMPLATE.md:10]:
```suggestion
<!-- List major points of changes here, so that the reviewers can have a bit more context while looking at your work! -->
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (eurunuela) [inline .github/PULL_REQUEST_TEMPLATE.md:37]: Shouldn't this last sentence be for the reviewers?

File: docs/contributorfile.rst
Comment (eurunuela) [inline docs/contributorfile.rst:112]:
```suggestion
A big challenge of software development is merging code accurately but without having to wait too much time.
```

File: docs/contributorfile.rst
Comment (eurunuela) [inline docs/contributorfile.rst:200]: I don't agree with this. Why would we want to merge some code that we know needs corrections? We should not merge code that contains errors.

Another thing is that from that PR we think of another PR with a different goal. Then it makes sense to work on a new PR after merging.

File: docs/contributorfile.rst
Comment (eurunuela) [inline docs/contributorfile.rst:202]: This is also a bit confusing. So, it's better to split the PR when assigning more than one label. But then you say the PR should have at least one label. It sounds like we're encouraging people to give more than one label with the "at least".

**Review by eurunuela — CHANGES_REQUESTED** [review]
My comments are not critical but I think some of the rephrasing @tsalo suggested should be included.

File: docs/contributorfile.rst
Comment (smoia) [reply docs/contributorfile.rst:200]: Let's see if the new rewording helps clarifying this point (commit to follow).

File: docs/contributorfile.rst
Comment (smoia) [reply docs/contributorfile.rst:202]: Same, let's see if it's better now.

**Comment by smoia** [comment]
@eurunuela do you think it's clearer now?

File: docs/contributorfile.rst
Comment (eurunuela) [reply docs/contributorfile.rst:216]: I think this one should be inlcuded.

File: docs/contributorfile.rst
Comment (eurunuela) [reply docs/contributorfile.rst:237]: I think this one should be included too.

File: docs/contributorfile.rst
Comment (eurunuela) [reply docs/contributorfile.rst:266]: This one should be included.

**Comment by eurunuela** [comment]
> @eurunuela do you think it's clearer now?

Sorry I didn't make it clearer before. I commented on the ones I think should be included.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (smoia) [reply .github/PULL_REQUEST_TEMPLATE.md:37]: Ah, no. This is for the user to clarify that they want a review while drafting (otherwise in the documentation we wrote not to do that)

**Comment by smoia** [comment]
@eurunuela now?

**Review by eurunuela — APPROVED** [review]
LGTM! Thank you!

**Comment by smoia** [comment]
@tsalo @RayStick if it's ok for you we can merge this in?

**Review by tsalo — APPROVED** [review]
I'm happy with the changes. Thank you.

**Comment by smoia** [comment]
@RayStick do you mind if we merge this?

**Comment by eurunuela** [comment]
> @RayStick do you mind if we merge this?

She already approved, so I guess she's happy 🤷🏻‍♂️

**Comment by eurunuela** [comment]
We should celebrate this!

**Comment by RayStick** [comment]
Great.  :tada:

**Comment by smoia** [comment]
:rocket: PR was released in [`2.3.0`](https://github.com/physiopy/phys2bids/releases/tag/2.3.0) :rocket:

**Pull request merged**
